fineract-provider/src/main/java/org/apache/fineract/portfolio/client/data/ClientData.java

```diff
@@ -34,82 +36,83 @@
 import org.apache.fineract.organisation.office.data.OfficeData;
 import org.apache.fineract.organisation.staff.data.StaffData;
 import org.apache.fineract.portfolio.address.data.AddressData;
-import org.apache.fineract.portfolio.collateralmanagement.domain.ClientCollateralManagement;
 import org.apache.fineract.portfolio.group.data.GroupGeneralData;
 import org.apache.fineract.portfolio.savings.data.SavingsAccountData;
 import org.apache.fineract.portfolio.savings.data.SavingsProductData;
 
 /**
  * Immutable data object representing client data.
  */
+@NoArgsConstructor
 @Getter
+@Setter
 public final class ClientData implements Comparable<ClientData>, Serializable {
 
-    private final Long id;
-    private final String accountNo;
-    private final String externalId;
-
-    private final EnumOptionData status;
-    private final CodeValueData subStatus;
-
-    private final Boolean active;
-    private final LocalDate activationDate;
-
-    private final String firstname;
-    private final String middlename;
-    private final String lastname;
-    private final String fullname;
-    private final String displayName;
-    private final String mobileNo;
-    private final String emailAddress;
-    private final LocalDate dateOfBirth;
-    private final CodeValueData gender;
-    private final CodeValueData clientType;
-    private final CodeValueData clientClassification;
-    private final Boolean isStaff;
-
-    private final Long officeId;
-    private final String officeName;
-    private final Long transferToOfficeId;
-    private final String transferToOfficeName;
-
-    private final Long imageId;
-    private final Boolean imagePresent;
-    private final Long staffId;
-    private final String staffName;
-    private final ClientTimelineData timeline;
-
-    private final Long savingsProductId;
-    private final String savingsProductName;
-
-    private final Long savingsAccountId;
-    private final EnumOptionData legalForm;
-    private final Set<ClientCollateralManagementData> clientCollateralManagements;
+    private Long id;
+    private String accountNo;
+    private String externalId;
```

Comment:
We should use ExternalId as a type here...

Reply:
Done!